packages/jsii-pacmak/lib/targets/dotnet/dotnetgenerator.ts

```diff
@@ -443,18 +437,36 @@ export class DotNetGenerator extends Generator {
         }
     }
 
-    // TODO: I uncovered an issue with optional parameters and ordering them
-    // They are currently not supported by the generator.
-    // In C#, optional parameters need to be declared after required parameters
-    // We could make changes to the parameters ordering in the jsii model to support them
-    // But then an optional parameter becoming non optional would create a mess
-    // https://github.com/awslabs/jsii/issues/210
+    /**
+     * Renders method parameters string
+     */
     private renderMethodParameters(method: spec.Method): string {
+        return this.renderParametersString(method.parameters);
+    }
+
+    /**
+     * Renders parameters string for methods or constructors
+     */
+    private renderParametersString(parameters: spec.Parameter[] | undefined): string {
         const params = [];
-        if (method.parameters) {
-            for (const p of method.parameters) {
-                const isOptionalPrimitive = this.isOptionalPrimitive(p) ? '?' : '';
-                const st = `${this.typeresolver.toDotNetType(p.type)}${isOptionalPrimitive} ${this.nameutils.convertParameterName(p.name)}`;
+        if (parameters) {
+            for (const p of parameters) {
+                let optionalPrimitive = '';
+                let optionalKeyword = '';
+                let type = this.typeresolver.toDotNetType(p.type);
+                if (p.optional) {
+                    if (this.isOptionalPrimitive(p)) {
+                        optionalPrimitive = '?';
+                        optionalKeyword = ' = null';
+                    } else {
+                        optionalKeyword = ' = null';
```

Comment:
The embedded if and else both set optionalKeyword to the same. Move this set to the parent if.

Reply:
Oops, this is a refactoring leftover. Removing